svc-aggregation/system/addaggregationsource.go

```diff
@@ -140,6 +141,12 @@ func (e *ExternalInterface) addAggregationSource(taskID, targetURI, reqBody stri
 	// else return the response
 	statusResp, statusCode, queueList := checkStatus(pluginContactRequest, addResourceRequest, cmVariants, taskInfo)
 	if statusCode == http.StatusOK {
+		_, errs := agmodel.GetPluginData(cmVariants.PluginID)
+		if errs == nil || (errs != nil && (errs.ErrNo() == errors.JSONUnmarshalFailed || errs.ErrNo() == errors.DecryptionFailed)) {
```

Comment:
Error check and error returned does not match, please update the condition to return different error responses based on the error encountered.

Reply:
modified the code